playground/backend/internal/code_processing/code_processing.go

```diff
@@ -93,14 +105,27 @@ func Process(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycl
 	}
 
 	// build executor for run step
-	exec = runBuilder.Build()
+	executor = runBuilder.Build()
 
 	// run
 	logger.Infof("%s: Run() ...\n", pipelineId)
-	runCmd := exec.Run(ctxWithTimeout)
-	go processCmd(runCmd, successChannel, errorChannel, dataChannel)
+	runCmd := executor.Run(ctxWithTimeout)
+
+	var runError bytes.Buffer
+	runOutput := &streaming.RunOutputWriter{Ctx: ctxWithTimeout, CacheService: cacheService, PipelineId: pipelineId}
```

Comment:
I am a little confused. I think `RunOutputWriter` fails if there has not been any output, right? what happens at the very begining when there's no output at all? what adds the very first output?